pkg/cmd/secvulnMaven.go

```diff
@@ -5,33 +5,299 @@
+		panic(err)
+	}
+
+	createPseudoMavenProject(mainPom)
+	fmt.Println("Pseudo maven project created for: " + mainPom.ArtifactId)
+
+	// Repeat the process for all dependencies with groupId dev.galasa
+	fmt.Println("Pseudo maven projects created for dependency chain of " + mainPom.ArtifactId + ":")
+	for len(toDoProjects) > 0 {
+
+		var index = len(toDoProjects) - 1
+		var artifactName = toDoProjects[index]
+
+		if artifactName == "com.jcraft.jsch" {
+			toDoProjects = removeItemFromArrayByIndex(toDoProjects, index)
+			continue
+		}
+
+		if bool := checkIfCompleted(artifactName); bool == true {
+			toDoProjects = removeItemFromArrayByIndex(toDoProjects, index)
+			continue
+		}
+
+		var version string
+		if version = getVersion(artifactName, mainPom); version == "" {
+			fmt.Printf("Unable to get version for artifact %s \n", artifactName)
+			panic(nil)
+		}
+		var currentPom Pom
+		var err error
+		for _, repo := range *secvulnMavenPomRepos {
+			currentPom, err = readPomFromRepo(repo, artifactName, version)
+			if currentPom.ArtifactId == artifactName {
+				break
+			}
+		}
+		if err != nil {
+			fmt.Printf("Could not find pom for artifact %s \n", artifactName)
+			panic(err)
+		}
+
+		createPseudoMavenProject(currentPom)
+
+		toDoProjects = removeItemFromArrayByIndex(toDoProjects, index)
+
+		fmt.Println("- " + artifactName)
+	}
+
+}
+
+func readPomFromUrl(url string) (Pom, error) {
+
+	var pom Pom
+
+	response, err := http.Get(url)
+	if err != nil {
+		return pom, err
+	}
+
+	defer response.Body.Close()
+
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return pom, err
+	}
+
+	err = xml.Unmarshal(body, &pom)
+	if err != nil {
+		return pom, err
+	}
+
+	return pom, nil
+
+}
+
+func readPomFromRepo(repo, artifactName, version string) (Pom, error) {
+
+	var url string
+	if repo == "https://galasadev-cicsk8s.hursley.ibm.com/main/maven/obr" {
+		url = fmt.Sprintf("%s/dev/galasa/%s/%s/%s-%s.pom", repo, artifactName, version, artifactName, version)
+
+	} else if repo == "https://repo.maven.apache.org/maven2" {
+		// Future proofing for artifacts like com.jcraft.jsch - can be improved
+		s := strings.Split(artifactName, ".")
+		var artifact string
+		for i := 0; i < len(s); i++ {
+			artifact += s[i] + "/"
+		}
+		url = fmt.Sprintf("%s/%s%s/%s-%s.pom", repo, artifact, version, s[len(s)-1], version)
+	} else {
+		fmt.Println("Invalid repositories provided to galasabld")
+		panic(nil)
+	}
+
+	return readPomFromUrl(url)
+
+}
+
+func createPseudoMavenProject(pom Pom) {
+
+	createDirectory(pom.ArtifactId)
+
+	createPom(pom)
+
+	completedProjects = append(completedProjects, pom.ArtifactId)
+}
+
+func createDirectory(artifactName string) {
+
+	if err := os.Mkdir(fmt.Sprintf("%s/%s", secvulnMavenParentDir, artifactName), os.ModePerm); err != nil {
+		fmt.Printf("Unable to create directory for artifact %s - %v", artifactName, err)
+	}
+
+	// Change into the sub-directory
+	os.Chdir(fmt.Sprintf("%s/%s", secvulnMavenParentDir, artifactName))
```

Comment:
dont use chdir in a program, you don't want the change the working directory ever.
use the full path the filename instead.

Reply:
done